src/tools/perf/lib/libperf.c

```diff
@@ -1749,42 +1790,14 @@ ucs_status_t ucx_perf_run(const ucx_perf_params_t *params,
         goto out;
     }
 
-    ucx_perf_test_init(perf, params);
-
-    if ((perf->send_allocator == NULL) || (perf->recv_allocator == NULL)) {
-        ucs_error("Unsupported memory types %s<->%s",
-                  ucs_memory_type_names[params->send_mem_type],
-                  ucs_memory_type_names[params->recv_mem_type]);
-        status = UCS_ERR_UNSUPPORTED;
-        goto out_free;
-    }
-
     if (params->api == UCX_PERF_API_UCT) {
-        if (perf->send_allocator->mem_type != UCS_MEMORY_TYPE_HOST) {
-            ucs_diag("UCT tests also copy one-byte value from %s memory to "
-                     "%s send memory, which may impact performance results",
-                     ucs_memory_type_names[UCS_MEMORY_TYPE_HOST],
-                     ucs_memory_type_names[perf->send_allocator->mem_type]);
-        }
-
-        if (perf->recv_allocator->mem_type != UCS_MEMORY_TYPE_HOST) {
-            ucs_diag("UCT tests also copy one-byte value from %s recv memory "
-                     "to %s memory, which may impact performance results",
-                     ucs_memory_type_names[perf->recv_allocator->mem_type],
-                     ucs_memory_type_names[UCS_MEMORY_TYPE_HOST]);
-        }
-    }
-
-    status = perf->send_allocator->init(perf);
-    if (status != UCS_OK) {
-        goto out_free;
-    }
-
-    if (perf->send_allocator != perf->recv_allocator) {
-        status = perf->recv_allocator->init(perf);
+        ucx_perf_global_init();
+        status = ucx_perf_allocators_init(perf, params);
```

Comment:
why do we need this change? we're not supposed to change UCT flow..

Reply:
I didn't change UCT flow but made some of the flow exclusive to UCT

Reply:
is this change really necessary? what happens if you don't change it?

Reply:
UCP will try to call `perf->send_allocator->init()` and will depend on the existence of allocators